crates/noirc_evaluator/src/ssa_refactor/opt/flatten_cfg.rs

```diff
@@ -329,13 +329,15 @@ impl<'f> Context<'f> {
     fn push_condition(&mut self, start_block: BasicBlockId, condition: ValueId) {
         let end_block = self.branch_ends[&start_block];
 
-        if let Some((_, previous_condition)) = self.conditions.last() {
+        let active_condition = if let Some((_, previous_condition)) = self.conditions.last() {
```

Comment:
`push_condition` is only called in `inline_branch`, so I think you'll need to add another enable_side_effects before the end block is inlined to restore the previous condition.

Reply:
Thanks! Now fixed.